### PR TITLE
feat: added library methods for conformance testing

### PR DESCRIPTION
**Pull request by shaffeeullah**

### CODE REVIEW

**Comment by shaffeeullah** [comment]
@ddelgrosso1 I still have to address the noted TODOs, but I would love an initial review on this.

**Comment by shaffeeullah** [comment]
Lint check is failing because these functions aren't called yet. There will be a separate PR for that implementation.

File: conformance-test/libraryMethods.ts [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {O_APPEND} from 'constants';
+import {file} from 'tmp';
+import {Bucket, File, HmacKey, Iam, Notification, Storage} from '../src';
+const fs = require('fs');
```
Comment (ddelgrosso1) [inline conformance-test/libraryMethods.ts:18]: Nitpicky but can this be changed to `import * as from 'fs';` just so there isn't mixed imports and requires.

**Comment by ddelgrosso1** [comment]
Took a look through and it looks like a solid start. Nothing stood out as note worthy to me.

**Review by ddelgrosso1 — APPROVED** [review]

**Review by tritone — APPROVED** [review]

**Pull request merged**